src/components/dashboard/ActionBarChart.js

```diff
@@ -0,0 +1,116 @@
+import React, { PureComponent } from 'react';
+import { withTranslation } from 'react-i18next';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import {
+  BarChart,
+  Bar,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+} from 'recharts';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core';
+import Loader from '../common/Loader';
+import Styles from '../../Styles';
+
+class ActionBarChart extends PureComponent {
+  static propTypes = {
+    classes: PropTypes.shape({
+      root: PropTypes.string.isRequired,
+      appBar: PropTypes.string.isRequired,
+      appBarShift: PropTypes.string.isRequired,
+      menuButton: PropTypes.string.isRequired,
+      hide: PropTypes.string.isRequired,
+      drawer: PropTypes.string.isRequired,
+      drawerPaper: PropTypes.string.isRequired,
+      drawerHeader: PropTypes.string.isRequired,
+      content: PropTypes.string.isRequired,
+      contentShift: PropTypes.string.isRequired,
+      developer: PropTypes.string.isRequired,
+      screenTitle: PropTypes.string.isRequired,
+    }).isRequired,
+    theme: PropTypes.shape({
+      palette: PropTypes.shape({
+        type: PropTypes.string.isRequired,
+        primary: PropTypes.arrayOf(PropTypes.string).isRequired,
+      }).isRequired,
+    }).isRequired,
+    history: PropTypes.shape({
+      replace: PropTypes.func.isRequired,
+    }).isRequired,
+    i18n: PropTypes.shape({
+      changeLanguage: PropTypes.func.isRequired,
+    }).isRequired,
+    t: PropTypes.func.isRequired,
+    actions: PropTypes.arrayOf(PropTypes.object),
+    spaces: PropTypes.arrayOf(PropTypes.object),
+  };
+
+  static defaultProps = {
+    actions: [],
+    spaces: [],
+  };
+
+  render() {
+    const { spaces, theme, t, actions } = this.props;
+    const {
+      palette: { primary, type },
+    } = theme;
+
+    if (!actions || _.isEmpty(actions) || !spaces || _.isEmpty(spaces)) {
```

Comment:
Here we will have an infinite loader if we have empty actions or spaces. How can we avoid this? Can't we use activity for the `Loader` and otherwise we show `No actions have been recorded.` message.